### PR TITLE
Development

### PR DESCRIPTION
**Pull request by MeqdamAlqudah**
- Sign up and log in pages
  - The user should be able to register in the app with full name, email and password (all mandatory).
  - The user can log into the app using email and password.
  - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).
  - Home page (catagories page)
  - When the user logs in, they are presented with the categories page.
  For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category. 
  - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.
  - Transactions page

- For a given category, the list of transactions is presented, ordered by the most recent.
  At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in 
 that category).
 - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
When the user clicks on the "Back" button (<), the user navigates to the home page.
"Add a new category" page

- The user fills out a form to create a new category, indicating their name and icon (both mandatory).
- The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
When the user clicks on the "Back" button (<), the user navigates to the home page.
"Add a new transaction" page

- The user fills out a form to create a new transaction with:
   - name (mandatory)
   - amount (mandatory)
   - categories (mandatory at least one)
The user click (or taps) the "Save" button to create the new category, and is taken to the transactions page for that category.
When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.
[video link](https://www.loom.com/share/1eaeb7e57caf4a698f8f63feb5353e0d)

### CODE REVIEW

File: app/views/deals/show.html.erb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    <%= image_tag @deal.category.icon %>
+    <p><%= @deal.name %><p>
+  </div>
+  <div>
+    <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.</p>
+  </div>
+  <hr>
+  <div>
+    <p>Transaction N<small>&#176;</small>3412000032</p>
+    <p><%= image_tag 'calendar.png' %><%= @deal.created_at %></p>
```
Comment (micheaol) [inline app/views/deals/show.html.erb:13]: According to the project requirement, please ensure that follow these [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), including:

* Colors
* Typography: font face, size, and weight.
* Layout: composition and space between elements.should follow these [design guidelines] for all the pages of the application.

File: app/views/deals/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+<%= render 'shared/header', title: 'DETAILS', image1: '', image3: 'back_arrow1.png', image2: '' %>
+<section class="details-card">
+  <div>
+    <%= image_tag @deal.category.icon %>
+    <p><%= @deal.name %><p>
+  </div>
+  <div>
```
Comment (micheaol) [inline app/views/deals/show.html.erb:7]: Please ensure that the pages that are not given a design should be created following the design guidelines of the other pages according to the project requirement.

**Review by micheaol — CHANGES_REQUESTED** [review]
Hi @MeqdamAlqudah , 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!


## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by uwadonat — COMMENTED** [review]
Hi **[MeqdamAlqudah](https://github.com/MeqdamAlqudah),** :hand: 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

### To highlight! :+1: 
- Your commit and pull request messages are very descriptive ✔️
- Your code is organized and clean  ✔️
almost

 ## Status: Required Changes ♻️

- [ ] Please try to cover most of features in your project in video presentation including the code you find more interesting and you are proud of.

- [ ] Check the comments under the review

## Optional

- [ ] Consider to not merge the pull request before reviewer approval.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**